vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/InputType.java

```diff
@@ -684,15 +704,44 @@ public static InputType dictionary(InputType key, InputType value) {
    * Create a new object type
    *
    * @param fields the names and types of the fields; duplicate names are not permitted
+   * @throws IllegalArgumentException if given a Stream with 0 elements, or a field key is empty
+   *     string
    */
   public static InputType object(Stream<Pair<String, InputType>> fields) {
-    return new ObjectInputType(fields);
+    // Sanity checking
+    List fieldsList = fields.collect(Collectors.toList());
```

Comment:
If you use `final var` here, the type will be preserved and then you won't need the casts later on.